### PR TITLE
perf: Various small performance improvements.

### PR DESCRIPTION
**Pull request by misteroneill**
These are a few performance improvements.

- Use `textContent` instead of `innerHTML` for text content.
- Construct `TextTrackSettings` content with HTML strings where possible because it's a bit faster than manual DOM construction.
- Do minimal DOM updates to time controls. This reduces our `timeupdate` footprint from ~2-2.5ms to ~1ms!

## Requirements Checklist
- [x] Feature implemented / Bug fixed
- [ ] Reviewed by Two Core Contributors

### CODE REVIEW

File: src/js/control-bar/time-controls/current-time-display.js [hunk 1]
```diff
@@ -62,12 +82,11 @@ class CurrentTimeDisplay extends Component {
   updateContent(event) {
     // Allows for smooth scrubbing, when player can't keep up.
     const time = (this.player_.scrubbing()) ? this.player_.getCache().currentTime : this.player_.currentTime();
-    const localizedText = this.localize('Current Time');
     const formattedTime = formatTime(time, this.player_.duration());
 
     if (formattedTime !== this.formattedTime_) {
       this.formattedTime_ = formattedTime;
-      this.contentEl_.innerHTML = `<span class="vjs-control-text">${localizedText}</span> ${formattedTime}`;
+      this.updateTextNode_();
```
Comment (gkatsev) [inline src/js/control-bar/time-controls/current-time-display.js]: we should probably wrap this in a rAF as well.

File: src/js/control-bar/time-controls/duration-display.js [hunk 2]
```diff
@@ -24,13 +26,18 @@ class DurationDisplay extends Component {
...
-    this.on(player, 'timeupdate', this.updateContent);
-    this.on(player, 'loadedmetadata', this.updateContent);
+    this.on(player, [
+      'durationchange',
+
+      // Also listen for timeupdate and loadedmetadata because removing those
+      // listeners could have broken dependent applications/libraries. These
+      // can likely be removed for 7.0.
+      'loadedmetadata',
+      'timeupdate'
```
Comment (gkatsev) [inline src/js/control-bar/time-controls/duration-display.js]: Noooooo. Did we forget to remove `timeupdate` in 6.x? T__T

File: src/js/tech/html5.js [hunk 3]
```diff
@@ -368,24 +368,26 @@ class Html5 extends Tech {
     // playback has started, which triggers the live display erroneously.
     // Return NaN if playback has not started and trigger a durationupdate once
     // the duration can be reliably known.
-    if (this.el_.duration === Infinity &&
```
Comment (gkatsev) [inline src/js/tech/html5.js]: is this section mostly formatting?

File: src/js/control-bar/time-controls/current-time-display.js [hunk 4]
```diff
@@ -24,7 +26,9 @@ class CurrentTimeDisplay extends Component {
   constructor(player, options) {
     super(player, options);
 
-    this.on(player, 'timeupdate', this.updateContent);
+    this.formattedTime_ = '0:00';
+    this.throttledUpdateContent = throttle(bind(this, this.updateContent), 100);
```
Comment (gkatsev) [inline src/js/control-bar/time-controls/current-time-display.js]: how come here it's 100 and below it's 25?

File: src/js/control-bar/time-controls/current-time-display.js (hunk 1 above)
Comment (gkatsev) [inline src/js/control-bar/time-controls/current-time-display.js]: And below as well.

File: src/js/control-bar/time-controls/duration-display.js (hunk 2 above)
Comment (misteroneill) [inline src/js/control-bar/time-controls/duration-display.js]: Yeah. 😢

File: src/js/tech/html5.js (hunk 3 above)
Comment (misteroneill) [inline src/js/tech/html5.js]: Yeah. It's going from:

```js
if (condition) {
  if (condition) {
    // Do stuff.
  }
}
```
to:
```js
if (condition && condition) {
  // Do stuff.
}
```

File: src/js/control-bar/time-controls/current-time-display.js (hunk 4 above)
Comment (misteroneill) [inline src/js/control-bar/time-controls/current-time-display.js]: Whoops. :)

File: src/js/tech/html5.js (hunk 3 above)
Comment (misteroneill) [inline src/js/tech/html5.js]: Not strictly perf-related, but I noticed it in searching for `timeupdate` listeners.

File: src/js/control-bar/time-controls/current-time-display.js [hunk 5]
```diff
@@ -39,19 +43,35 @@ class CurrentTimeDisplay extends Component {
...
+   * Updates the "current time" text node with new content using the
+   * contents of the `formattedTime_` property.
+   *
+   * @private
+   */
+  updateTextNode_() {
+    if (this.textNode_) {
+      this.contentEl_.removeChild(this.textNode_);
+    }
+    this.textNode_ = document.createTextNode(` ${this.formattedTime_}`);
```
Comment (gkatsev) [inline src/js/control-bar/time-controls/current-time-display.js]: `updateTextNode_` gets called from `createEl`, which in turn, gets called during component init which happens before we set `this.formattedTime_`. We should either have a default value for when `this.formattedTime_` isn't set, or we need to call `updateTextNode_` in the end of the constructor.
This applies to the currentTimeDisplay and duration display and whatever else uses this `updateTextNode_`.
undefined gif

Comment (misteroneill) [inline src/js/control-bar/time-controls/current-time-display.js]: I think the former is preferable. I'll update. Can't believe I somehow didn't notice this...

**Pull request merged**